base_miner/UCF/train_ucf.py

```diff
@@ -0,0 +1,370 @@
+# This script was adapted from the DeepfakeBench training code,
+# originally authored by Zhiyuan Yan (zhiyuanyan@link.cuhk.edu.cn)
+
+# Original: https://github.com/SCLBD/DeepfakeBench/blob/main/training/train.py
+
+# BitMind's modifications include adding a testing phase, changing the 
+# data load/split pipeline to work with subnet 34's image augmentations
+# and datasets from BitMind HuggingFace repositories, quality of life CLI args,
+# logging changes, etc.
+
+import os
+import sys
+import argparse
+from os.path import join
+import random
+import datetime
+import time
+import yaml
+from tqdm import tqdm
+import numpy as np
+from datetime import timedelta
+from copy import deepcopy
+from PIL import Image as pil_image
+from pathlib import Path
+import gc
+
+import torch
+import torch.nn as nn
+import torch.nn.parallel
+import torch.backends.cudnn as cudnn
+import torch.utils.data
+import torch.optim as optim
+from torch.utils.data.distributed import DistributedSampler
+import torch.distributed as dist
+from torch.utils.data import DataLoader
+
+from optimizor.SAM import SAM
+from optimizor.LinearLR import LinearDecayLR
+
+from trainer.trainer import Trainer
+from detectors import DETECTOR
+from metrics.utils import parse_metric_for_print
+from logger import create_logger, RankFilter
+
+from huggingface_hub import hf_hub_download
+
+# BitMind imports (not from original Deepfake Bench repo)
+from bitmind.dataset_processing.load_split_data import load_datasets, create_real_fake_datasets
+from bitmind.image_transforms import base_transforms, random_aug_transforms
+from bitmind.constants import DATASET_META, FACE_TRAINING_DATASET_META
+
+parser = argparse.ArgumentParser(description='Process some paths.')
+parser.add_argument('--detector_path', type=str,
+                    default=os.getcwd() + '/config/ucf.yaml',
+                    help='path to detector YAML file')
+parser.add_argument("--train_dataset", nargs="+")
+parser.add_argument("--test_dataset", nargs="+")
+parser.add_argument('--faces_only', dest='faces_only', action='store_true', default=False)
+parser.add_argument('--no-save_ckpt', dest='save_ckpt', action='store_false', default=True)
+parser.add_argument('--no-save_feat', dest='save_feat', action='store_false', default=True)
+parser.add_argument("--ddp", action='store_true', default=False)
+parser.add_argument('--local_rank', type=int, default=0)
+parser.add_argument('--task_target', type=str, default="", help='specify the target of current training task')
+args = parser.parse_args()
+torch.cuda.set_device(args.local_rank)
+print(f"torch.cuda.device(0): {torch.cuda.device(0)}")
+print(f"torch.cuda.get_device_name(0): {torch.cuda.get_device_name(0)}")
+
+def ensure_backbone_is_available(logger,
+                                 weights_dir='./weights/',
+                                 model_filename='xception_best.pth',
+                                 hugging_face_repo_name='bitmind/ucf/'):
```

Comment:
[Suggestion] config file for paths would be good, esp since these are shared across different classes